src/Octoshift/MessageEnhancer.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OctoshiftCLI;
+
+internal static class MessageEnhancer
+{
+    private static readonly Dictionary<string, string> _map = new()
+    {
+        ["not have the correct permissions to execute"] = "Please check that (a) you are a member of the `{0}` organization, " +
+                                                          "(b) you are an organization owner or you have been granted the migrator role and " +
+                                                          "(c) your personal access token has the correct scopes. " +
+                                                          "For more information, see https://docs.github.com/en/migrations/using-github-enterprise-importer/preparing-to-migrate-with-github-enterprise-importer/managing-access-for-github-enterprise-importer."
+    };
+
+    public static bool IsEnhanceable(string message) => GetKeyForMessage(message) is not null;
+
+    public static string Enhance(string message, Func<string, string, string> messageFactory = null)
+    {
+        messageFactory ??= (originalMessage, improvedMessage) => $"{originalMessage}{improvedMessage}";
+
+        var key = GetKeyForMessage(message);
+        return key is null ? message : messageFactory(message, _map[key]);
+    }
+
+    private static string GetKeyForMessage(string message) => _map.Keys.SingleOrDefault(key => message.Contains(key, StringComparison.OrdinalIgnoreCase));
+}
```

Comment:
This seems like it's getting kinda convoluted. I'll give more feedback when I have a bit more time, but my gut reaction is there's gotta be a more straightforward way to do this. Maybe a custom exception that inherits from OctoshiftCliException.

Reply:
Well, I agree that for just this one case it's a bit of an overkill, but my main goal for recommending this was to cover future use-cases too. It seems like we are going to have quite a few error message improvements like this and the problem is not only the exception message but also those `if`s and `when`s to look for those key sentences in error messages all over different places. With this approach we can aggregate everything in one place.
One other approach would be to create some sort of constants holder class to store those enhanced messages and also the key sentences that we are going to look for (in `when` clauses).

Reply:
I think this design will kinda work, but the most complex part is "variables" inside the messages. In this case, we want to interpolate the organization name, but in other messages, it might be something else that we want to interpolate.

If we're going to design a generic solution to use in future use cases, I think we need to figure that out.

Reply:
That makes sense. But then the class calling `Enhance` needs to know ahead of time what error it's going to get back right, because it has to know what the placeholders will be? Does that make sense? I might not be explaining clearly!

Reply:
well we know what error message we are going to improve right? the improved message is going to be a value in the dictionary which will contain those placeholders. Now if an error message is enhanceable (which means if any of the predefined keys in or dictionary were found in the message text), then we are going the get the enhanced version and replace the placeholders with required info. Of course we need to carefully choose the keys in such a way that they don't exist if more than one message and then with the lambda (which is the dynamic part) we can compose the end result message.